### PR TITLE
Use args of logging functions for lazy formatting

### PR DESCRIPTION
**Pull request by TimoWilken**
Logging functions (like `debug`, `info`, `error`) can take extra arguments, which are substituted into the message using printf-like syntax (`%s` and the like). This formatting is only done if the given logging level is actually enabled, giving some minor speed up in case debug logging is disabled. It also makes the code cleaner, as e.g. `debug(format("%s %s", a, b))` can be written as `debug("%s %s", a, b)` instead.

This patch also updates unit tests to work with the built-in logging convention.

Also includes fixes for the minor problem where `err = execute(cmd); debug(err, cmd)` would not output `cmd`; only `err` (the exit code) without explanation. I've changed this to output both `cmd` and `err` with a more descriptive message.

Tested and working.

### CODE REVIEW

File: tests/test_parseRecipe.py [hunk 1]
```diff
@@ -57,8 +57,8 @@
 class Recoder(object):
   def __init__(self):
     self.buffer = ""
-  def __call__(self, s):
-    self.buffer += s
+  def __call__(self, s, *a):
+    self.buffer += s % a
```
Comment (ktf) [inline tests/test_parseRecipe.py:61]: can you do `format(s, **kwds) % *a` instead, so that we can also use the keywords?

**Review by ktf — COMMENTED** [review]
Maybe we can keep the current keyword based functionality?

File: tests/test_parseRecipe.py (hunk 1 above)
Comment (TimoWilken) [reply tests/test_parseRecipe.py:61]: I'm not a big fan of overriding the builtin `format` (which does something else), though `s % (kwds or a)` would work, I suppose...

The problem is that the built-in logger doesn't have that functionality, I think. It uses keywords for special situations (e.g. to tell it that it should print exception info attached to the log line). There might be a way to retrofit that, but it's likely a bit hacky, if not impossible.

Comment (ktf) [reply tests/test_parseRecipe.py:61]: For the record, the reason why we have an override for the format method is that at the time we had to support python 2.5. If you want, feel free to replace all of them with `str.format`. That said, please, do not go and replace the current "keyword based" solution everywhere with positional formatting.

Given you replaced the keyword functionality only in an handful of places let's merge this.

**Review by ktf — APPROVED** [review]
For the record, the reason why we have an override for the format method is that at the time we had to support python 2.5. If you want, feel free to replace all of them with str.format. That said, please, do not go and replace the current "keyword based" solution everywhere with positional formatting.

Given you replaced the keyword functionality only in an handful of places let's merge this.

**Pull request merged**